ts/features/itwallet/common/utils/itwClaimsUtils.ts

```diff
@@ -350,43 +350,52 @@ export const getCredentialExpireDays = (
   return differenceInCalendarDays(expireDate, Date.now());
 };
 
+type GetCredentialStatusOptions = {
+  /**
+   * Number of days before expiration required to mark a credential as "EXPIRING".
+   */
+  expiringDays?: number;
+  /**
+   * Check the expiration using the JWT `exp` claim, not the credential itself.
+   */
+  checkJwtExpiration?: boolean;
+};
+
 /**
- * Returns the expire status of a {@link ParsedCredential}, taking into account the **expiration date only**.
- * Use {@link getCredentialStatus} to also check the status attestation.
- * @param credential the parsed credential claims
- * @param expiringDays the number of days required to mark a credential as "EXPIRING"
- * @returns "VALID" if the credential is valid, "EXPIRING" if there are less than {expiringDays} days left until the expiry day, "EXPIRED" if the expiry date has passed
+ * Get the overall status of the credential, taking into account
+ * the status attestation if present and the credential's own expiration date.
+ *
+ * @param credential the stored credential
+ * @param options see {@link GetCredentialStatusOptions}
+ * @returns ItwCredentialStatus
  */
-export const getCredentialExpireStatus = (
-  credential: ParsedCredential,
-  expiringDays: number = 14
+export const getCredentialStatus = (
+  credential: StoredCredential,
+  options: GetCredentialStatusOptions = {}
 ): ItwCredentialStatus | undefined => {
-  const expireDays = getCredentialExpireDays(credential);
+  if (credential.storedStatusAttestation?.credentialStatus === "invalid") {
+    return "expired";
+  }
+
+  const { checkJwtExpiration, expiringDays = 14 } = options;
 
-  if (expireDays === undefined) {
+  const expireDate = checkJwtExpiration
+    ? credential.jwt.expiration
+    : getCredentialExpireDate(credential.parsedCredential);
+
+  if (expireDate === undefined) {
     return undefined;
   }
 
+  const expireDays = differenceInCalendarDays(expireDate, Date.now());
+
   return expireDays > expiringDays
```

Comment:
What do you think about replacing this ternary with a simpler if statement like this one for clarity?
It would make this part more readable.

```
  if (expireDays > expiringDays) {
    return "valid";
  }

  if (expireDays > 0) {
    return "expiring";
  }

  return "expired";
  ```

Reply:
Addressed in 2ab6063a8e2e45a48eca27de28d728d24d38c42b.